cpp/src/arrow/adapters/orc/adapter_test.cc

```diff
@@ -730,4 +733,30 @@ TEST_F(TestORCWriterSingleArray, WriteListOfMap) {
   AssertArrayWriteReadEqual(array, array, kDefaultSmallMemStreamSize * 10);
 }
 
+class TestORCWriterMultipleWrite : public ::testing::Test {
+ public:
+  TestORCWriterMultipleWrite() : rand(kRandomSeed) {}
+
+ protected:
+  random::RandomArrayGenerator rand;
+};
+
+TEST_F(TestORCWriterMultipleWrite, MultipleWritesIntField) {
```

Comment:
Nice test!

Could you also add one to verify that we get a sensible error if you write a table but then try to write a second table with a different schema?

Reply:
Yes, I can try do do this. Though, I will need to modify the code I think.

Reply:
Done